manifests/znode.pp

```diff
@@ -2,16 +2,18 @@
   $ensure = '',
   $assosiation = ''
 ) {
-  
+
+  $znode_check = "/bin/bash -c \"if [[ \\\"$(echo 'stat /${name}' | /opt/zookeeper/bin/zkCli.sh 2>&1 1>/dev/null)\\\" == 'Node does not exist'* ]]; then exit 1; else exit 0; fi\""
```

Comment:
1.) rely on bash's `[[]]`'s exit code, remove extraneous inversion of exit using `if` and instead use `!=`,  avoid multi backslashes. 
```
$znode_check = "/bin/bash -c '[[ $(echo \"stat /${name}\" | /opt/zookeeper/bin/zkCli.sh 2>&1 1>/dev/null) != \"Node does not exist\"* ]]'"
```

2.) use posix shell's case to get wildcard match
```
$znode_check = "case $(echo 'stat /${name}' | /opt/zookeeper/bin/zkCli.sh 2>&1 1>/dev/null) in 'Node does not exist'*) true;; *) false ;; esac"
```
3.) use grep, rely on it's  exit code
```
$znode_check = "echo 'stat /${name}' | /opt/zookeeper/bin/zkCli.sh 2>&1 1>/dev/null | grep -vq 'Node does not exist'"
```
4.) write a command line utility, install it on the box with a file resource and call it from the unless hook. 

I prefer number 4,  if you feel like messing with it.